fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/IndexingBase.java

```diff
@@ -231,9 +251,57 @@ private boolean shouldClearExistingIndexEntries(@Nonnull IndexState indexState,
         }
     }
 
+    @Nonnull
+    public void setFallbackMode() {
+        forceStampOverwrite = true; // must overwrite a previous indexing method's stamp
+    }
+
+    @Nonnull
+    private CompletableFuture<Void> setIndexingTypeOrThrow() {
+        return getRunner().runAsync(context -> openRecordStore(context).thenCompose(store -> {
+            Transaction transaction = store.getContext().ensureActive();
+            byte[] stamp = getIndexingTypeStamp();
```

Comment:
Also, did you fix here?

Reply:
If I remember correctly, you've asked me to change the usage of `buildCommitRetryAsync` to `runAsync` - what am I missing?